Code/src/main/java/nl/utwente/group10/ui/handlers/ConnectionCreationManager.java

```diff
@@ -0,0 +1,91 @@
+package nl.utwente.group10.ui.handlers;
+
+import java.util.HashMap;
+import java.util.Map;
+
+
+import java.util.Optional;
+
+import javafx.geometry.Point2D;
+import javafx.scene.Parent;
+import nl.utwente.group10.ui.CustomUIPane;
+import nl.utwente.group10.ui.components.Connection;
+import nl.utwente.group10.ui.components.ConnectionAnchor;
+import nl.utwente.group10.ui.components.ConnectionLine;
+import nl.utwente.group10.ui.components.InputAnchor;
+import nl.utwente.group10.ui.components.OutputAnchor;
+
+public class ConnectionCreationManager {
+
+	CustomUIPane pane;
+
+	/**
+	 * Touch points have an ID associated with each specific touch point, this
+	 * is the ID associated with the Mouse.
+	 */
+	public static final Integer MOUSE_ID = 0;
+	/**
+	 * Maps an (Touch or Mouse) ID to a line, used to keep track of what touch
+	 * point is dragging what line.
+	 */
+	private Map<Integer, Connection> connections;
+	
+	
+	public ConnectionCreationManager(CustomUIPane pane){
+		this.pane = pane;
+		connections = new HashMap<Integer, Connection>();
+	}
+	
+
+	public Connection createConnectionWith(int id, ConnectionAnchor anchor) {
+		Connection newConnection = null;
+		if(anchor instanceof OutputAnchor){
+			newConnection = new Connection((OutputAnchor) anchor);
+		} else if(anchor instanceof InputAnchor){
+			newConnection = new Connection((InputAnchor) anchor);
+		}
```

Comment:
`new Connection(anchor)` lukt inderdaad niet, maar nu gooi je een NPE als het geen van beide is, dat lijkt me ook niet echt mooi - misschien een `createConnectionWith` voor `InputAnchor`s en één `createConnectionWith` voor `OutputAnchor`s?